staging/src/k8s.io/apimachinery/pkg/util/proxy/transport.go

```diff
@@ -232,6 +233,15 @@ func (t *Transport) rewriteResponse(req *http.Request, resp *http.Response) (*ht
 		defer gzw.Close()
 		writer = gzw
 	// TODO: support flate, other encodings.
+	case "deflate":
```

Comment:
This whole response rewriting thing was a huge mistake, I'm not super excited to add bells and whistles. @deads2k thoughts? I guess this is at least a small, targeted change...

Please update the TODO statement and add a test.

Reply:
Updated
Thanks